python/desc/imsim/trim.py

```diff
@@ -99,14 +103,24 @@ def get_object_entries(self, chip_name, radius=0.18, sort_magnorm=True):
                     number of sersic objects for minsource application)
 
         """
+        self.trimmer.logger.debug("computing object offsets from %s center",
+                                  chip_name)
         ra0, dec0 = self.compute_chip_center(chip_name)
         seps = degrees_separation(ra0, dec0, self._ra, self._dec)
-        index = np.where(seps < radius)
+        if chip_name in self.trimmer.drawn_objects:
+            not_drawn = [not x in self.trimmer.drawn_objects[chip_name]
```

Comment:
If this run starts from a checkpoint, this code omits objects that have already been drawn from the objects that are passed to the downstream object processing code.  In addition to speeding up that processing, the per sensor memory usage can be substantially reduced.

Reply:
Old comment I originally typed:

```
I'm having a hard time understanding the 'chip_name in self.trimmer.drawn_objects' since in the gs_intepreter the list is made out of  uniqueIds here: self.drawn_objects.add(gsObject.uniqueId).  How does the chip_name get in?

I think it is related to the fact that the sensor name is in the checkpoint file?
```

OK, I think I understand this now. The trimmer drawn_objects is different:

self.drawn_objects[detname] = ckpt['drawn_objects']

I think maybe the names of the two variables should be different for clarity.  Perhaps one could be drawn_objects_on_sensor or something?